core/src/main/java/hudson/init/TaskMethodFinder.java

```diff
@@ -116,8 +116,8 @@ protected void invoke(Method e) {
      */
     private Object lookUp(Class<?> type) {
         if (type==Jenkins.class || type==Hudson.class)
-            return Jenkins.getInstanceOrNull();
-        Jenkins j = Jenkins.getInstanceOrNull();
+            return Jenkins.getInstance();
+        Jenkins j = Jenkins.getInstance();
         if (j!=null) {
```

Comment:
Obviously the second invocation should not have been replaced, since you are still checking for null. Given the presence in the `init` package I suspect that `getInstanceOrNull` is correct in both invocations.